src/net/http/UploadFile.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "UploadFile.hpp"
+#include "ToStream.hpp"
+
+#include "Operation/Operation.hpp"
+#include "OutputStreamHandler.hxx"
+#include "Progress.hpp"
+#include "Request.hxx"
+#include "util/ScopeExit.hxx"
+#include "system/ConvertPathName.hpp"
+
+#include <cassert>
+
+void
+Net::UploadFile(CurlGlobal &curl, const char *url,
+                const char *username, const char *password,
+                http_post* formpost,  Json::ParserOutputStream *parser,
```

Comment:
Change to OutputStream - so it is in the hand of the caller :) - In a previous state I changed it directly to a json value - but this should the caller do, of course ...
'... a pointer?': I learned a lot with the google coding style: There has forbidden a non-const reference (from my point of view for a simular reason like the proscription of a const_cast inside a low level function) - but XCSoar use this very often -and so I will do it in the same way...

Reply:
I know little of the Google coding style (I know they forbid C++ exceptions which I disagree with); but what's the point of disallowing non-const references, but allowing non-const pointers? Pointers are less safe than references because they are nullable, everything else is the same. Doesn't make sense.